engine/src/main/java/org/terasology/world/generator/plugin/WorldGeneratorListener.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.world.generator.plugin;
+
+import org.terasology.world.generation.FacetProviderListener;
+
+/**
+ * A category of {@link Class}es that would like to know about certain aspects of world generation, but do not seek
+ * to <em>influence</em> world generation.
+ * <br/>
+ * Current subclasses are:
+ * {@link FacetProviderListener}
+ */
+public interface WorldGeneratorListener {
```

Comment:
This relates somewhat to my overall comment that you might as well run the listeners at the end once all the data is available, but why would you need other sorts of WorldGenerationListener, and even if this did have multiple sub-interfaces, what would be the point of this interface, which doesn't have any methods? If there's nothing that you can do with both a FacetProviderListener and (hypothetically) a RasterizerListener (not that I think that actually makes much sense), where would you need to deal with something that's either one or the other, but you don't care which?

Reply:
As above I guess I was cargo cult-ing again. I'll rename ```FacetProviderListener``` to ```WorldGeneratorListener``` and make it the one interface.